REFERENCES.md

```diff
@@ -0,0 +1,329 @@
+
+# Using TypeScript Project References with ts-loader and Webpack
+
+Project References were added to TypeScript in 3.0. The benefits of using project references include:
+
+* Better code organisation
+* Logical separation between components
+* Faster build times
+
+If you are using TypeScript in your web project you can also use project references to improve your code and build workflow. This article describes some of the ways to set up your project to use references. I am using ts-loader to transpile the TypeScript code to JavaScript and Webpack to bundle code.
+
+An example repo using the configuration above is available at the link below:
+
+[https://github.com/appzuka/project-references-example](https://github.com/appzuka/project-references-example)
+
+There are 2 stages to using project references in your project:
+
+1. Configure and build the project references
+1. Setup your codebase to consume the compiled projects
+
+### Configure and build the project references
+
+This stage just involves following the directions from the TypeScript documentation:
```

Comment:
As I understand it from the "Using ts-loader to build project references" section, following the approach prior to that is an alternative approach that could be used and also probably *not* the one people will likely go for. `ts-loader` building references is just going to be simpler (my presumption)

As such, is it worth calling out here that these docs are going to outline 2 approaches more explicitly?

Reply:
I wanted to show how to configure using tsc --build because you need to get that part right or nothing is going to work.  It is also helpful to check that it is working with tsc independently if you are having problems.  Once it is working I believe most people would use ts-loader.

I'll amend the docs to make it clear I am not proposing 2 alternatives.